pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -451,6 +455,80 @@ if [ "$action" = dry-build ]; then
     extraBuildFlags+=(--dry-run)
 fi
 
+if [ "$action" = list-generations ]; then
+    if [ ! -L "$profile" ]; then
+        echo "No profile \`$(basename $profile)' found" >&2
+        exit 1
+    fi
+
+    generation_from_dir() {
+        generation_dir="$1"
+        generation_base="$(basename "$generation_dir")" # Has the format "system-123-link" for generation 123
+        no_link_gen="${generation_base%-link}"  # remove the "-link"
+        echo ${no_link_gen##*-} # remove everything before the last dash
+    }
+    describe_generation(){
```

Comment:
In a different programming language I would much rather have this function generic and then have a transform, depending on if the output is `json` or not. Only I don't know how to do this in bash...